src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -225,6 +226,10 @@ public async Task InitializeUIAffinitizedServicesAsync(IAsyncServiceProvider asy
 
             Logger.Log(FunctionId.Run_Environment,
                 KeyValueLogMessage.Create(m => m["Version"] = FileVersionInfo.GetVersionInfo(typeof(VisualStudioWorkspace).Assembly.Location).FileVersion));
+
+            // Initialize task-list in BG.
+            var taskListService = this.Services.SolutionServices.ExportProvider.GetExports<VisualStudioTaskListService>().Single().Value;
+            taskListService.Start(this);
```

Comment:
💭 We really need to provide a better way to delay-start services.

Reply:
We are still delaying everything until the window is actually visible.  This just hooks our stuff up to start listening for that.

Reply:
Right, my complaint was not about the specific time where this occurs, but rather the direct coupling to VisualStudioWorkspaceImpl.

Reply:
I view things as effectively directly coupled anyways.  Just in a very baroque fashion.  :)  This code also seems to have just so many layers of unnecessary indirection.  I've reduced it down from all that complexity and overhead to just two classes and this call to start everything up.  Seems simple and sane and straightforward.  It doesn't move us back, and it gets cleaned up in a few months/releases anyways.  I don't see the harm here.